08_AdvancedTypes/UniversalTablePrinter.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System.Reflection;
+using System.Text;
+
+public class UniversalTablePrinter
+{
+    private readonly IEnumerable<object> _data;
+    private const int _columnLength = 20;
+
+    public UniversalTablePrinter(IEnumerable<object> data)
+    {
+        _data = data;
+    }
+    public void PrintToConsole()
+    {
+        object firstElement = _data.First();
+        if (firstElement is not null)
+        {
+            Type type = firstElement.GetType();
+            PropertyInfo[] props = type.GetProperties();
+
+            //add table columns name
+            StringBuilder columns = new StringBuilder();
+            foreach (PropertyInfo prop in props)
+            {
+                columns.Append($"{prop.Name,-_columnLength}|");
```

Comment:
You could go over each column and understand it minimum length, if we have any column with more characters than column length it's going to break the table, but it would also require traversing the items more than one time